### PR TITLE
Validate adding groups to conversations

### LINKED ISSUE #137: Improve UX when no user has been added as particpant
When adding a user group as a participant, it is not guaranteed that any new user is added to the conversation, for example, if the user group is empty or if all of its members are already part of the conversation. In such a case, `count: 0` is returned by the request, no message is being shown and the dialog is simply closed without giving the active user any feedback on their action.

### PR DESCRIPTION
**Pull request by TimWolla**
- [x] The English phrases will need some copy-editing, they sound off to me, but I can't pinpoint my mistake.
- [ ] Commit 2 introduces a behavioral change which I consider helpful, because it prevents mistakes. It can easily be dropped if undesired, though.

------

Resolves #137

### CODE REVIEW

**Comment by TimWolla** [comment]
> I get a wrong error message (`Every member within the given user groups already participates in this conversation.`) when I add groups that are not actually addable to conversations. This happens when there is no group that can be added to a conversation (so I can select any group in the drop-down menu). Probably this was also wrong before.

This is fixed by 04684f400b214db9436497b645519f79bd92eedd.

**Comment by TimWolla** [comment]
> I find the behavior misleading for the user, if I enter all users manually + the group itself, that I get an error message that all members are already in the conversation. Wouldn't it be nicer for the Ux here if this was silently ignored and didn't lead to an error message, especially if this error message only appears if each specified group would not add any users?

I don't agree here. If the user adds a user group where all group members are already part of the conversation or added at the same time as the group then that's most likely not intended by the user. Preventing submission allows the user to either remove the group in question or replace it by the proper group. Simply ignoring such a group can be pretty confusing as well. The user will see a confirm message (which is better than the current situation), but afterwards nothing visible has changed. The user might rightfully assume a software bug then.



> Also, empty groups also lead to such a strange error message. Here you should add the indicator that the group is empty and therefore cannot be added.

Ideally such groups would not even be offered. I'm not sure whether it is worthwhile detecting this case. It would require the administrator to make the group addable to conversations, but leave it empty. Also the error message technically is *not wrong*.

File: language/de.xml [hunk 1]
```diff
@@ -103,6 +103,8 @@
 		<item name="wcf.conversation.participants.error.duplicate"><![CDATA[{$errorData[username]} ist bereits Teilnehmer dieser Konversation.]]></item>
 		<item name="wcf.conversation.participants.error.canNotUseConversation"><![CDATA[{$errorData[username]} besitzt nicht die notwendigen Berechtigungen, um an einer Konversation teilnehmen zu dürfen.]]></item>
 		<item name="wcf.conversation.participants.error.doesNotAcceptConversation"><![CDATA[{$errorData[username]} möchte nicht an Konversationen teilnehmen.]]></item>
+		<item name="wcf.conversation.participants.error.emptyGroup"><![CDATA[Die Mitglieder aller angegebener Gruppen sind bereits Teilnehmer dieser Konversation.]]></item>
+		<item name="wcf.conversation.participants.error.intersects"><![CDATA[{$errorData[username]} ist ebenfalls als sichtbarer Teilnehmer eingetragen.]]></item>
```
Comment (dtdesign) [inline language/de.xml:107]:
```suggestion
		<item name="wcf.conversation.participants.error.intersects"><![CDATA[{$errorData[username]} ist bereits als sichtbarer Teilnehmer eingetragen.]]></item>
```

File: language/de.xml [hunk 2]
```diff
@@ -103,6 +103,8 @@
 		<item name="wcf.conversation.participants.error.duplicate"><![CDATA[{$errorData[username]} ist bereits Teilnehmer dieser Konversation.]]></item>
 		<item name="wcf.conversation.participants.error.canNotUseConversation"><![CDATA[{$errorData[username]} besitzt nicht die notwendigen Berechtigungen, um an einer Konversation teilnehmen zu dürfen.]]></item>
 		<item name="wcf.conversation.participants.error.doesNotAcceptConversation"><![CDATA[{$errorData[username]} möchte nicht an Konversationen teilnehmen.]]></item>
+		<item name="wcf.conversation.participants.error.emptyGroup"><![CDATA[Die Mitglieder aller angegebener Gruppen sind bereits Teilnehmer dieser Konversation.]]></item>
```
Comment (dtdesign) [inline language/de.xml:106]:
```suggestion
		<item name="wcf.conversation.participants.error.emptyGroup"><![CDATA[Alle Mitglieder der angegebenen Gruppen sind bereits Teilnehmer dieser Konversation.]]></item>
```

File: language/en.xml [hunk 3]
```diff
@@ -103,6 +103,8 @@
 		<item name="wcf.conversation.participants.error.duplicate"><![CDATA[{$errorData[username]} is already a participant of this conversation.]]></item>
 		<item name="wcf.conversation.participants.error.canNotUseConversation"><![CDATA[{$errorData[username]} is not allowed to participate in a conversation.]]></item>
 		<item name="wcf.conversation.participants.error.doesNotAcceptConversation"><![CDATA[{$errorData[username]} does not want to be added in conversations.]]></item>
+		<item name="wcf.conversation.participants.error.emptyGroup"><![CDATA[Every member within the given user groups already participates in this conversation.]]></item>
```
Comment (dtdesign) [inline language/en.xml:106]:
```suggestion
		<item name="wcf.conversation.participants.error.emptyGroup"><![CDATA[All members of the given user groups already participate in this conversation.]]></item>
```

File: language/en.xml [hunk 4]
```diff
@@ -103,6 +103,8 @@
...
+		<item name="wcf.conversation.participants.error.intersects"><![CDATA[{$errorData[username]} was also added as a visible participant.]]></item>
```
Comment (dtdesign) [inline language/en.xml:107]:
```suggestion
		<item name="wcf.conversation.participants.error.intersects"><![CDATA[{$errorData[username]} was already added as a visible participant.]]></item>
```

**Comment by dtdesign** [comment]
I agree with @TimWolla on the error message for "empty" group invites. Eventually the user has to manually remove such a superfluous group, but this is inconvenient at worst while being fully transparent about the behavior.

**Pull request merged**